src/clientSideScene/CameraControls.ts

```diff
@@ -213,7 +149,77 @@ export class CameraControls {
     this.update(true)
   }
 
-  constructor(isOrtho = false, domElement: HTMLCanvasElement) {
+  throttledEngCmd = throttle((cmd: EngineCommand) => {
+    this.engineCommandManager.sendSceneCommand(cmd)
+  }, 1000 / 30)
+
+  throttledUpdateEngineCamera = throttle((threeValues: ThreeCamValues) => {
+    const cmd: EngineCommand = {
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'default_camera_look_at',
+        ...convertThreeCamValuesToEngineCam(threeValues),
+      },
+    }
+    this.engineCommandManager.sendSceneCommand(cmd)
+  }, 1000 / 15)
+
+  lastPerspectiveCmd: EngineCommand | null = null
+  lastPerspectiveCmdTime: number = Date.now()
+  lastPerspectiveCmdTimeoutId: number | null = null
+
+  sendLastPerspectiveReliableChannel = () => {
+    if (
+      this.lastPerspectiveCmd &&
+      Date.now() - this.lastPerspectiveCmdTime >= lastCmdDelay
+    ) {
+      this.engineCommandManager.sendSceneCommand(this.lastPerspectiveCmd, true)
+      this.lastPerspectiveCmdTime = Date.now()
+    }
+  }
+
+  throttledUpdateEngineFov = throttle(
+    (vals: {
+      position: Vector3
+      quaternion: Quaternion
+      zoom: number
+      fov: number
+      target: Vector3
+    }) => {
+      const cmd: EngineCommand = {
+        type: 'modeling_cmd_req',
+        cmd_id: uuidv4(),
+        cmd: {
+          type: 'default_camera_perspective_settings',
+          ...convertThreeCamValuesToEngineCam({
+            ...vals,
+            isPerspective: true,
+          }),
+          fov_y: vals.fov,
+          ...calculateNearFarFromFOV(vals.fov),
+        },
+      }
+      this.engineCommandManager.sendSceneCommand(cmd)
+      this.lastPerspectiveCmd = cmd
+      this.lastPerspectiveCmdTime = Date.now()
+      if (this.lastPerspectiveCmdTimeoutId !== null) {
+        clearTimeout(this.lastPerspectiveCmdTimeoutId)
+      }
+      this.lastPerspectiveCmdTimeoutId = setTimeout(
+        this.sendLastPerspectiveReliableChannel,
+        lastCmdDelay
+      ) as any as number
+    },
+    1000 / 30
+  )
+
+  constructor(
+    isOrtho = false,
+    domElement: HTMLCanvasElement,
+    engineCommandManager: EngineCommandManager
```

Comment:
I didn't mess around with the order too much, but yeah I'd say `engineCommandManager` is the primary dependency, but it's is kinda obvious when you start trying to tease it apart, it's just used the most.

I don't think this is dependency injection. I might be wrong, but I'm just being explicit with prop drilling effectively, I know there's definitely not going to be circles if I have to pass it in.

I'd say a lot of this is my naviety with classes, I'm really not too keen to do the whole class inheritance complexity, but singletons always seemed like a nice pattern since I at least thought you we didn't need to worry about class instances or how to get access to themo where we need them (adding it to context or similar) it can always be imported.

Where I've gone astray is making multiple singletons that are not isolated, as in they want to call each other's methods, this could easy be solved by making this all into one giant class, but was trying to keep things somewhat separate by their purpose. I'm sure there's a better pattern, but in the meantime being a little less flippant with how they're setup and how they call each other's methods should stop this from biting us in the arse in the immediate future.